### PR TITLE
Introduce priority class to Kyma components

### PR DESCRIPTION
**Pull request by ebensom**
**Description**

On Kyma runtimes, we must ensure prioritized scheduling of Kyma components over customer workloads. We have seen multiple occurrences where the users overloaded the cluster with their workloads thus creating a fight between kyma components and user workloads scheduling. It caused service degradation and created issues during kyma upgrades as surge pods couldn't be scheduled.

**Related issue(s)**
https://github.tools.sap/kyma/backlog/issues/1207

### CODE REVIEW

**Review by a-thaler — DISMISSED** [review]

**Review by Demonsthere — APPROVED** [review]
/meow

**Review by franpog859 — DISMISSED** [review]

File: resources/rafter/templates/pre-upgrade-delete-vs-job.yaml [hunk 1]
```diff
@@ -109,3 +109,6 @@ spec:
           defaultMode: 420
           name: pre-upgrade-delete-vs-rafter-minio
         name: scripts
+    {{- if .Values.global.priorityClassName }}
```
Comment (tgorgol) [inline resources/rafter/templates/pre-upgrade-delete-vs-job.yaml:112]: I think we should also add the priorityClassName to [this](https://github.com/kyma-project/kyma/blob/8eb85671fec6676e117bfa4967a16ed5b5121f54/resources/rafter/charts/controller-manager/charts/minio/templates/ensure-secret-job.yaml#L56) job in the rafter chart.

File: resources/cluster-essentials/templates/priorityclass.yaml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+{{- if .Values.global.priorityClassName }}
+apiVersion: scheduling.k8s.io/v1
+kind: PriorityClass
+metadata:
+  name: {{ .Values.global.priorityClassName }}
+value: {{ .Values.global.priorityClassValue }}
+globalDefault: false
+description: "Default scheduling priority of Kyma installer. Must not be blocked by unschedulable user or Kyma workloads."
```
Comment (Tomasz-Smelcerz-SAP) [inline resources/cluster-essentials/templates/priorityclass.yaml:8]: The description is wrong. It's default PriorityClass for all Kyma components but the Installer. Installer has it's own class named `kyma-installer`

**Review by sayanh — DISMISSED** [review]

File: resources/cluster-essentials/templates/priorityclass.yaml (hunk 2 above)
Comment (ebensom) [reply resources/cluster-essentials/templates/priorityclass.yaml:8]: Copy-paste power! :)

**Review by tgorgol — APPROVED** [review]

**Review by franpog859 — APPROVED** [review]

**Review by marcobebway — APPROVED** [review]

**Review by a-thaler — APPROVED** [review]

**Review by dariadomagala — APPROVED** [review]

**Review by piotrmiskiewicz — APPROVED** [review]

**Review by adamwalach — APPROVED** [review]

**Review by mjakobczyk — APPROVED** [review]

**Pull request merged**